GoogleForCreators/web-stories-wp
# Implemented new design for tab view

## Context

This implements the new design as per [Figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=1009%3A4323) for both the icon tab list in the library and text tab list in the inspector.

This also updates the active tab indicator and the warning icon sizing and color next to the checklist as well as heavy modifications to the focus styling of the different tabs (icon or text).

Finally, this updates Storybook to reflect both current usages of the generalized tab view.

## User-facing changes

| | Before | After |
|-|-|-|
| Elements, Keyboard | ![element-tabs-keyboard-before](https://user-images.githubusercontent.com/637548/109718487-6c1a8700-7b75-11eb-8b6b-b08f767195c7.gif) | ![element-tabs-keyboard](https://user-images.githubusercontent.com/637548/109718498-6de44a80-7b75-11eb-9a3d-99c3fc298a1e.gif) |
| Elements, Mouse | ![element-tabs-mouse-before](https://user-images.githubusercontent.com/637548/109718539-79d00c80-7b75-11eb-800e-962195b1779f.gif) | ![element-tabs-mouse](https://user-images.githubusercontent.com/637548/109718550-7ccafd00-7b75-11eb-87d6-23188ae25f38.gif) |
| Design, Keyboard | ![design-tabs-keyboard-before](https://user-images.githubusercontent.com/637548/109718589-86546500-7b75-11eb-82dd-1c06416316ee.gif) | ![design-tabs-keyboard](https://user-images.githubusercontent.com/637548/109718594-881e2880-7b75-11eb-85ed-2889f2fbc21b.gif) |
| Design, Mouse | ![design-tabs-mouse-before](https://user-images.githubusercontent.com/637548/109718645-92402700-7b75-11eb-94fa-d9345b429843.gif) | ![design-tabs-mouse](https://user-images.githubusercontent.com/637548/109718660-95d3ae00-7b75-11eb-861b-3b531c8d5aa5.gif) |

## Todo

- [x] Update padding for tab buttons
- [x] Update active indicator for tab buttons
- [x] Allow inspector tabs to be unjustified
- [x] Update styling of tab extra-icon if exists
- [x] Fix focus styling

## Testing Instructions

### QA

This PR can be tested by following these steps:

1. Select a tab in the element library panel
2. Observe the tab changes
3. Select a tab in the design panel tab list
4. Observe the tab changes

### UAT




## Checklist




---

Fixes #6212
